shared/js/background/tracker-utils.js

```diff
@@ -198,6 +198,19 @@ function truncateReferrer (referrer, target) {
     return modifiedReferrer
 }
 
+/**
+ * Checks if a tracker is a first party by checking entity data
+ * @param {string} trackerUrl
+ * @param {string} siteUrl
+ * @param {object} request
+ * @returns {boolean}
+ */
+function isFirstPartyByEntity (trackerUrl, siteUrl, request) {
+    if (!trackerUrl) return false
+    const trackerData = trackers.getTrackerData(trackerUrl, siteUrl, request)
+    return trackerData ? trackerData.firstParty : utils.isFirstParty(trackerUrl, siteUrl) // fall back on hostname check if trackers is null
```

Comment:
Can we please rename `isFirstParty` helper to `isSameTopLevelDomain`, so that it's super clear what it does?